### PR TITLE
Rc

### PR DESCRIPTION
**Pull request by WouterVandeWiele**
Adding Microsoft Azure AD authentication.  #746

` authentmake test` failures:
```shell
...
=== RUN   TestAuthProviderConfigSetupProvider
--- PASS: TestAuthProviderConfigSetupProvider (0.00s)
FAIL
coverage: 76.4% of statements
FAIL    github.com/pocketbase/pocketbase/core   0.043s
...
=== RUN   TestPointer
--- PASS: TestPointer (0.00s)
PASS
coverage: 97.5% of statements
ok      github.com/pocketbase/pocketbase/tools/types    0.002s  coverage: 97.5% of statements
?       github.com/pocketbase/pocketbase/ui     [no test files]
FAIL
make: *** [Makefile:5: test] Error 1
```

### CODE REVIEW

File: core/settings.go [hunk 1]
```diff
@@ -33,12 +33,13 @@ type Settings struct {
...
-	GitlabAuth   AuthProviderConfig `form:"gitlabAuth" json:"gitlabAuth"`
-	DiscordAuth  AuthProviderConfig `form:"discordAuth" json:"discordAuth"`
-	TwitterAuth  AuthProviderConfig `form:"twitterAuth" json:"twitterAuth"`
+	GoogleAuth      AuthProviderConfig `form:"googleAuth" json:"googleAuth"`
+	FacebookAuth    AuthProviderConfig `form:"facebookAuth" json:"facebookAuth"`
+	GithubAuth      AuthProviderConfig `form:"githubAuth" json:"githubAuth"`
+	GitlabAuth      AuthProviderConfig `form:"gitlabAuth" json:"gitlabAuth"`
+	DiscordAuth     AuthProviderConfig `form:"discordAuth" json:"discordAuth"`
+	TwitterAuth     AuthProviderConfig `form:"twitterAuth" json:"twitterAuth"`
+	MicrosoftAdAuth AuthProviderConfig `from:"microsoftAdAuth" json:"microsoftAdAuth"`
```
Comment (ganigeorgiev) [inline core/settings.go:42]: I think we can remove the "Ad" suffix from the provider name and all other places, aka. let's keep it just "MicrosoftAuth".

**Review by ganigeorgiev — CHANGES_REQUESTED** [review]

File: core/settings.go (hunk 1 above)
Comment (ollema) [inline core/settings.go:42]: is `from:` a typo here? all other providers user `form:`

Comment (WouterVandeWiele) [reply core/settings.go]: oops. Should be fixed now.

**Comment by WouterVandeWiele** [comment]
Changed the microsoftAuth names to azureAuth.

Also reordered the expected result in `core/settings_test.go` - `TestNamedAuthProviderConfigs`. Now the test failures from `make test`, mentioned in earlier, are gone.

**Comment by ganigeorgiev** [comment]
@WouterVandeWiele Thank you. I've merged and squashed the PR locally in https://github.com/pocketbase/pocketbase/commit/5298543ce486b0e4fe53dc1fc9a38b02313e22c5.

At the end, I've decided to keep the provider name "Microsoft" for simplicity. Other than "Windows's Live ID" I couldn't find any other OAuth2 services (and it seems that it is a legacy as far as I understand reading https://techcommunity.microsoft.com/t5/microsoft-entra-azure-ad-blog/cleaning-up-the-azuread-and-microsoft-account-overlap/ba-p/245105 and other related resources). Additionally Microsoft Azure AD seems to support both cases - Microsoft personal accounts as well as Azure Active Directory accounts (Firebase also seems to use only "Microsoft" as label in their UI and documentations https://firebase.google.com/docs/auth/android/microsoft-oauth).

I've also decided to add support for custom provider components loaded through the `provider.js` config file, which should result in better UX since now we can show specialized help-blocks or hide unnecessary fields (eg. the user api endpoint for the microsoft provider).

I've tested the integration with a developer account and it seems to work fine. If you stumble on something or don't agree with the changes, feel free to let me know I'll revisit them.

**Comment by WouterVandeWiele** [comment]
I concur on the naming.

The `optionsComponent` is also a great addition. 👍 
It's more customizable then the old way while keeping a clean interface IMHO.

Great work, thanks for adding the feature.

**Pull request closed without merging**